### PR TITLE
Update for safe startline parsing PR

### PR DESCRIPTION
**Pull request by benaadams**
Change for PR https://github.com/dotnet/aspnetcore/pull/20885 (unmerged)

/cc @halter73

### CODE REVIEW

File: src/BenchmarksApps/Kestrel/PlatformBenchmarks/BenchmarkApplication.HttpConnection.cs [hunk 1]
```diff
@@ -310,15 +323,16 @@ public void OnHeader(ReadOnlySpan<byte> name, ReadOnlySpan<byte> value)
 
             public void OnHeadersComplete(bool endStream)
                 => RequestHandler.OnHeadersComplete(endStream);
+            public void OnStartLine(HttpVersionAndMethod versionAndMethod, TargetOffsetPathLength targetPath, Span<byte> startLine)
```
Comment (halter73) [inline src/BenchmarksApps/Kestrel/PlatformBenchmarks/BenchmarkApplication.HttpConnection.cs:326]: nit:

```suggestion

            public void OnStartLine(HttpVersionAndMethod versionAndMethod, TargetOffsetPathLength targetPath, Span<byte> startLine)
```

**Comment by halter73** [comment]
@sebastienros The startline PR has been merged to master. Should we merge this now?

**Comment by sebastienros** [comment]
I think it's ok as the change was in aspnet. But let me test it tomorrow first.

**Comment by benaadams** [comment]
Currently Platform tests will break when aspnet is updated? (The regular ones should be fine as is internal)

**Pull request merged**